### PR TITLE
Desktop navigation animation improvements.

### LINKED ISSUE #17985: Story desktop performance low hanging fruits
Addressing a few easy fixes:

- [x] Blurred background should have a fadein animation
- [ ] Pages should not overlap during animations, it looks really bad since they have an opacity
- [x] Video should not be paused and rewinded during the animation, but after, so the animation doesn't look too "jumpy"
- [ ] Improving the easing curves

### PR DESCRIPTION
**Pull request by gmajoulet**
(_Based on @addieachan's PR #18408_)

Before: https://stamp-desktop-nav-before.firebaseapp.com/examples/s20/body-painting/index.html
After: https://stamp-desktop-nav-after.firebaseapp.com/examples/s20/body-painting/index.html

This PR fixes many issues with the desktop navigation animation:
- N-2 page used to disappear right away from the screen

N-2 page now translates out of the screen

- N+1 page would instantly appear and overlap with the transitioning active page

N+2 page is positioned correctly and translates to its N+1 position when needed, without overlapping any other page

- active page was positioned with the `[active]` attribute, and N+1/N-1 pages with other attributes that were set later. Animations weren't triggered at the exact same time and it looked very janky

All pages are now positioned against a new `i-amphtml-desktop-position` attribute, that is set in the same `mutate` callback to ensure they're all set in the same browser frame

- Opacity animation curves improvements

The active page becomes visible faster, and the N-1 page disappears a bit faster, to drag the attention on the new active page

Fixes #17985

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-desktop.css [hunk 1]
```diff
@@ -154,36 +140,67 @@ amp-story[standalone][desktop] {
   left: 0 !important;
 }
 
+/**
+ * Page positions and animations.
+ */
+
+[desktop] amp-story-page.i-amphtml-element {
+  transform: scale(0.9) translateX(calc(250% + 192px)) translateY(0%) !important;
```
[the remark this answers is not in the document]
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-desktop.css]: done

File: extensions/amp-story/1.0/amp-story.js [hunk 2]
```diff
@@ -1072,29 +1067,56 @@ export class AmpStory extends AMP.BaseElement {
...
+        list.push({page: this.getPageById(minusTwoId), position: -2});
+      }
     }
 
-    const nextPageId = targetPage.getNextPageId(
-        /* opt_isAutomaticAdvance */ false);
-    if (nextPageId) {
-      this.nextPage_ = this.getPageById(nextPageId);
-      setAttributeInMutate(this.nextPage_, Attributes.NEXT);
+    const plusOneId = targetPage.getNextPageId(false /** isAutomaticAdvance */);
```
[the remark this answers is not in the document]
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js]: done

File: extensions/amp-story/1.0/amp-story.js [hunk 3]
```diff
@@ -994,7 +994,9 @@ export class AmpStory extends AMP.BaseElement {
           oldPage.setState(PageState.NOT_ACTIVE);
 
           // Indication that this should be offscreen to left in desktop view.
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story.js]: nit: update this comment.

File: extensions/amp-story/1.0/amp-story.js [hunk 4]
```diff
@@ -994,7 +994,9 @@ export class AmpStory extends AMP.BaseElement {
...
-          setAttributeInMutate(oldPage, Attributes.VISITED);
+          this.getPageIndex(oldPage) < pageIndex ?
+            setAttributeInMutate(oldPage, Attributes.VISITED) :
+            removeAttributeInMutate(oldPage, Attributes.VISITED);
```
Comment (Enriqe) [inline extensions/amp-story/1.0/amp-story.js]: If `Attributes.VISITED` is just deciding whether the pages are hidden off-screen to the right or left, maybe we should change the name. e.g. `Attributes.HIDE_LEFT`.

Also, have you tried if this works with RTL?

Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story.js]: See new commit :))

**Comment by Enriqe** [comment]
RTL and LTR look great! :-) Thanks!!

**Pull request merged**